packages/core/src/routes/interaction/verifications/identifier-payload-verification.ts

```diff
@@ -81,27 +81,48 @@ const verifySocialIdentifier = async (
   return { key: 'social', connectorId: identifier.connectorId, userInfo };
 };
 
-const verifySocialIdentityInInteractionRecord = async (
-  { connectorId, identityType }: SocialIdentityPayload,
+const verifySocialVerifiedIdentifier = async (
+  payload: SocialVerifiedIdentifierPayload,
   ctx: WithLogContext,
   interactionRecord?: AnonymousInteractionResult
 ): Promise<VerifiedEmailIdentifier | VerifiedPhoneIdentifier> => {
   const log = ctx.createLog(`Interaction.SignIn.Identifier.Social.Submit`);
-  log.append({ connectorId, identityType });
+  log.append(payload);
+
+  const { connectorId } = payload;
 
   // Sign-In with social verified email or phone requires a social identifier in the interaction result
   const socialIdentifierRecord = interactionRecord?.identifiers?.find(
     (entity): entity is SocialIdentifier =>
       entity.key === 'social' && entity.connectorId === connectorId
   );
 
-  const verifiedSocialIdentity = socialIdentifierRecord?.userInfo[identityType];
+  assertThat(socialIdentifierRecord, new RequestError('session.connector_session_not_found'));
+
+  // Verified Email Payload
+  if ('email' in payload) {
+    const { email } = payload;
+    assertThat(
+      socialIdentifierRecord.userInfo.email === email,
+      new RequestError('session.connector_session_not_found')
```

Comment:
nit: Should we throw more specific error message instead of `connector_session_not_found`?